### PR TITLE
[WIP] Adding the ability to optionally change the CSS class(es) applied to …

### PR DESCRIPTION
**Pull request by rhamilto**
…registry-imagestream-listing so that Bootstrap/PatternFly styles can be used instead of the default.  This is so that in the context of origin-web-console, registry-imagestream-listing can have similar styling to the tables used throughout origin-web-console.  Note:  a few additional overrides are required in origin-web-console, but those will be added in the PR that consumes the changes in this PR.

screen shot 2017-08-02 at 4 50 12 pm

### CODE REVIEW

**Comment by stefwalter** [comment]
@petervo Could you review this?

@spadgett Could you add petervo and @thomasmckay as contributors so we can request reviews in the sidebar?

File: views/imagestream-listing.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<table class="listing-ct">
+<table ng-class="classNames ? '{{classNames}}' : 'listing-ct'">
```
Comment (petervo) [inline views/imagestream-listing.html:1]: I think a lot of other styles in here that you probably want to keep reference listing-ct. Either those need to change, or this should keep listing-ct even when classNames is present.

**Review by petervo — CHANGES_REQUESTED** [review]

**Comment by spadgett** [comment]
> @spadgett Could you add petervo and @thomasmckay as contributors so we can request reviews in the sidebar?

@dobbymoodge Can you help with this?

**Comment by spadgett** [comment]
@stefwalter @petervo Is there a reason we're not just the Patternfly table styles directly in the widgets?

**Comment by petervo** [comment]
@spadgett because that would totally break the styling in the registry. Alternative to this is to remove the table tag from the widget and make these directives

```
restrict: 'A',
```

instead of 'E'. Then each project can add their own table classes.

File: views/imagestream-listing.html (hunk 1 above)
Comment (rhamilto) [reply views/imagestream-listing.html:1]: listing-ct can still be passed in as an optional class.  In the origin-web-console, the listing-ct classes actually break our desired layout, hence this approach; this gives the consumers of the repo direct control over which styles are used while maintaining backward compatibility.

Comment (petervo) [reply views/imagestream-listing.html:1]: @rhamilto have you actually tried it? For me without the class a bunch of other styles don't apply. Is the plan to just have overrides there?

Comment (rhamilto) [reply views/imagestream-listing.html:1]: I have.  The [screenshot above](https://github.com/openshift/registry-image-widgets/pull/18#issue-247518924) utilizes the change in the context of origin-web-console.  In the case of the console, we appear to not be utilizing all of the functionality of the directive, so it's not a problem.  My ultimate aim is to simply fix https://github.com/openshift/origin-web-console/issues/1849, which this change enables.

**Pull request closed without merging**